### ISSUE TITLE
Child data set label and icons not showing up

### ISSUE THREAD

**Issue by cjh1** [OpenChemistry/tomviz#679]
If I apply the "Binary Threshold" I see the following:

![empty](https://cloud.githubusercontent.com/assets/1031053/19360883/5dbed712-914e-11e6-8496-4ebd30caad2b.png)

@cquammen Are you seeing this on master too? Its totally possible I broke this somehow.

**Comment by cjh1** [comment]
I don't seem to be able to add operators to them either ( child data sources that is ).

**Comment by cquammen** [comment]
I cannot reproduce on master - everything seems to work as expected.

Two comments:

1). When it works correctly, the name/icon doesn't appear until after the operator completes *and* you click on the item. Not ideal.

2). Check the output at the console. My guess is there was an error and no child data set was generated. In that case, the name/icon will not appear at all. Again, not ideal.

**Comment by cjh1** [comment]
The only thing I am seeing in the console is

```
Warning, array does not have Fortran order, making deep copy and fixing...
...done.
```

**Comment by cquammen** [comment]
Hmm, that looks like it completed just fine.

**Comment by cryos** [comment]
I took a look and see it complete, and that it is blank while processing, but shows up after. This should be looked at, but I can't reproduce your error here.

**Comment by cjh1** [comment]
I recompiled using RelWithDebInfo ( I was running Debug before ) and see it update when the processing is complete, looks like some sort of timing issue, I can take look.

**Comment by cjh1** [comment]
So #710 fixes my issue where the child data source was never showing up, I now see what everyone else was seeing ( it appears when the operation is complete ). So this behavior is a result of us creating the row in the table before we have a data source to populate it, so it appears blank until the child data source has been set. Now we could resolve this in by delaying the creation of the row until the data source is available. The problem with this is it is not consistent with our handling on results ( they appear up front ) although we could change that. I guess another option would be to create a "dummy" data source that would be used to render that table during the execution of the operator as we only really need the "filename". Thoughts?

**Comment by cryos** [comment]
I would favor probably go with delayed creation - the new operator is there, and once it completes, and there is data, we should then see what it is. Not sure how much utility there is in the promise of data, especially when we have an operator that we can mark as executing in the UI.

**Comment by cquammen** [comment]
Note that the child data source name should be available from the JSON.

**Comment by cryos** [comment]
Even if so you can't do anything with the data until it is available, wouldn't it be more intuitive to show it once it is available? I guess the flip side is if you know it will take a while, and want to chain an operator off of a child data set before it is ready, but would the pipeline function as expected? If we do show it before it is ready we should grey it out to make it clear the data is not really available, but I think it would be most intuitive to see it after it is created by the operator upon completion.

**Comment by cquammen** [comment]
I don't have a preference either way, implementation was just easier the way it is now.

I think it could be nice to have a place-holding indicating that something is coming. It could be greyed out to show that it isn't ready, and struck through if the production of the child data source fails. But either way is fine with me.

**Comment by cryos** [comment]
We should not show the child data sets until the operator has completed, let's work on that as the ideal model. When we rerun the pipeline we should grey out any objects that are invalidated, I think they are replaced once the operator at that stage has been rerun.

**Comment by cjh1** [comment]
Fixed by #812
